corona-warn-app/cwa-app-android
# CCL config and booster update mechanism (EXPOSUREAPP-11559)

https://github.com/corona-warn-app/cwa-app-tech-spec/blob/proposal/logic-engine/docs/spec/dcc-ccl.gen.md#api-for-updating-the-ccl-configuration-from-cdn

### What's in this PR: 
- Refactoring of Booster Notifications
- Scheduler that handles updates for the CCLConfig and BoosterNotificationRules and then triggers the calculation of `DccWalletInfo`
- Test Menu Item to force update the CCL configuration

### How to test Force-Update:
- Use MOCK-CLOUD
- Test Menu -> CCL Configuration & WalletInfo -> Force-Update CCL Configuration
- CCL Configuration and Booster rules should be updated
- DccWalletInfo Calculation should be performed

### Other things to test
- CCL Configuration & WalletInfo should be updated on app start when last update was >24h ago
- Worker that does an update & DccWalletInfo calculation daily should be scheduled

### What's left to do: 
- [x] Call DccWalletUpdateTrigger after the update to trigger the calculation (in `CCLConfigurationUpdater`)
- [x] Call CCLConfigRepository (once it is merged) in the `CCLConfigurationUpdater`
- [x] Call `BoosterNotificationService.notifyIfNecessary()` from the CCL Layer after every DCCWalletInfo calculation
- [x] Create more tests for `CCLConfigurationUpdater` 
- [x] Create more tests for `BoosterNotificationService`
